Library/Homebrew/extend/os/linux/dependency_collector.rb

```diff
@@ -12,7 +12,8 @@ class DependencyCollector
 
   sig { params(related_formula_names: T::Set[String]).returns(T.nilable(Dependency)) }
   def gcc_dep_if_needed(related_formula_names)
-    return unless DevelopmentTools.system_gcc_too_old?
+    # Brewed GCC is required for libgcc_s.so.1 if either the host Glibc or GCC is too old.
+    return unless DevelopmentTools.build_system_too_old?
```

Comment:
It seems like this risks making the preferred GCC a dependency of `glibc`, which we don't want, I think. But I guess local testing revealed this to not be the case?

Reply:
I wasn't able to test this PR fully because of `Error: maximum tree depth (10) exceeded calculating gcc@5 dependency tree!` (https://github.com/Homebrew/brew/pull/13733#issuecomment-1227704994) which affects `master` even without this PR.

Reply:
We definitely don't want `gcc` to be a dependency of `glibc` - it should only depend on `linux-headers`.  Maybe we can add `return if related_formula_names.include?(GLIBC)` below this so that this always returns if the formula is `glibc`.

Reply:
@sjackman Going to need you to dig in a bit harder on these sorts of issues, please. I don't even really have a Linux machine to test these on. If the value for that variable needs changed: consider changing it as part of this PR.

Reply:
https://docs.docker.com/desktop/install/mac-install/
I do all of my testing on Linux using Docker Desktop for macOS.

Reply:
The circular dependencies in Homebrew/core were resolved by (I believe)…
- https://github.com/Homebrew/homebrew-core/pull/109173
- https://github.com/Homebrew/homebrew-core/pull/109186
- https://github.com/Homebrew/homebrew-core/pull/109189